### DISCUSSION TITLE
RENOVATE_ONBOARDING_CONFIG is not respected

### DISCUSSION THREAD

**Discussion by hiaik** [renovatebot/renovate#32168] [Request Help]
### How are you running Renovate?

self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab (saas)

### Please tell us more about your question or problem

I am running renovate of version 38.132.3 (image ghcr.io/renovatebot/renovate:38.132.3).

I pass `RENOVATE_ONBOARDING_CONFIG` as env variable and it is not respected for onboarding MR. Incorrect value for config is still used.

### Logs (if relevant)

In the env variable I have
```
`RENOVATE_ONBOARDING_CONFIG: '{"$$schema": "https://docs.renovatebot.com/renovate-schema.json", "extends": ["gitlab><my project path>"] }'`
```

I use `gitlab>` as my project is on gitlab saas.

and in debug log I can see it is found:
```
DEBUG: CLI config
       "config": {}
DEBUG: Env config
       "config": {
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ],
         "repositoryCache": "enabled",
...
         "onboardingConfig": {
           "$schema": "https://docs.renovatebot.com/renovate-schema.json",
           "extends": ["gitlab><my project path>"]
         },
         "optimizeForDisabled": true,
...
       }
DEBUG: Combined config
       "config": {
...
         "onboardingConfig": {
           "$schema": "https://docs.renovatebot.com/renovate-schema.json",
           "extends": ["gitlab><my project path>"]
         },
...
```

`gitlab>` matches here -> all good.

Later in log I can see:
```
...
DEBUG: Onboarding PR does not exist (repository=...)
DEBUG: Checking for a default Renovate preset which can be used. (repository=...)
DEBUG: Checking for preset: local><my project path> (repository=...)
DEBUG: Found preset local><my project path> - using it in onboarding config (repository=...)
DEBUG: onboarding config (repository=...)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["local><my project path>"]
       }
```

and it results in onborading MR with `local>` and not `gitlab>` as I expect.

Please clarify on what is wrong?

**Comment by RahulGautamSingh** [comment]
This is expected behavior if you have saved the preset in a repo named `renovate-config`.
What happens is that first we look for any default presets hosted by user or their org on the same platform as the onboarding repo
eg. if you are on `github` then we will check if you have any repo named `renovate-config` present. Incase it is we automatically include this preset in you onboarding config without you needing to add it in the `RENOVATE_ONBOARDING_CONFIG` env variable.

`local` is used instead of `gitlab` because we only add this preset in your config it is on the same platform. [^1]

[^1]: https://docs.renovatebot.com/config-presets/#grouporganization-level-presets

**Reply by hiaik** [reply]
thanks @RahulGautamSingh 

We do have https://gitlab.com/.../renovate-config/-/blob/develop/default.json?ref_type=heads

Thanks for docs link, it is missing that important part you explained
```
local is used instead of gitlab because we only add this preset in your config it is on the same platform
```

That said, it means in our case when onboarding repo and renovate-config (default.json) are on the same gitlab platform, `local>` is fine.

**Reply by rarkins** [reply]
You said you're using a mend-hosted app. There is no mend app for GitLab.com. Please correct your original description if you are in fact self-hosting.

**Reply by hiaik** [reply]
> You said you're using a mend-hosted app. There is nonmetal for GitLab.com. Please correct your original description if you are in fact self-hosting.

@rarkins Done, apologies for the incorrect input

**Comment by rarkins** [comment]
Which Mend app are you using?

**Reply by hiaik** [reply]
> Which Mend app are you using?

fixed in description for self-hosted

**Comment by hiaik** [comment]
Hi @RahulGautamSingh 

> What happens is that first we look for any default presets hosted by user or their org on the same platform as the onboarding repo
eg. if you are on github then we will check if you have any repo named renovate-config present. Incase it is we automatically include this preset in you onboarding config without you needing to add it in the RENOVATE_ONBOARDING_CONFIG env variable. 

+ https://docs.renovatebot.com/config-presets/#grouporganization-level-presets

```
On platforms that support nested groups (e.g. GitLab), Renovate will check for this repository at each level of grouping, from nearest to furthest, and use the first one it finds.
```

I understood that doc as:
1. if I have config in `gitlab.com/org/group1/renovate-config`
2. when I ran renovate onboarding for `gitlab.com/org/group2/repo`, I expected renovate to look for `renovate-config` within all groups within `/viator` namespace. But it did not happen right now on our side. 

Does that mean it only finds `renovate-config` for the same group -> so for `org/group2/repo` it will not find config from `org/group1/renovate-config` ?

Docs says `at each level of grouping, from nearest to furthest`... I thought it should search for all groups from the parental one /org/ in my case.
